### PR TITLE
[KeyVault] Add Status Methods to Query Backup and Restore Operations

### LINKED ISSUE #13718: Allow status of backup and restore operations to be queried
To facilitate long-running backup and restore operations, provide idiomatic APIs to query status. This may be on the `KeyVaultBackupClient` directly, or on public operation classes.

### PR DESCRIPTION
**Pull request by mccoyp**
Closes #13718.

Adds methods to `KeyVaultBackupClient` for checking backup and restore operation status. We may want to create a new polling class to make getting job IDs (that are necessary for these methods) easier to obtain.

### CODE REVIEW

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py [hunk 1]
```diff
@@ -93,3 +93,35 @@ def begin_selective_restore(self, blob_storage_uri, sas_token, folder_name, key_
...
+        :rtype: BackupOperation
+        """
+        return self._client.full_backup_status(
+            vault_base_url=self._vault_url,
+            job_id=job_id,
+            cls=BackupOperation._wrap_generated,
+            **kwargs
+        )
+
+    def restore_status(self, job_id, **kwargs):
```
[the remark this answers is not in the document]
Comment (mccoyp) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py:113]: Those names definitely make more sense. I had replicated the names of the generated methods but will change that 👍

File: README.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ Last stable versions of packages that have been provided for usage with Azure an
 
 ### Management: New Releases
 A new set of management libraries that follow the [Azure SDK Design Guidelines for Python](https://azure.github.io/azure-sdk/python/guidelines/) are now available. These new libraries provide a number of core capabilities that are shared amongst all Azure SDKs, including the intuitive Azure Identity library, an HTTP Pipeline with custom policies, error-handling, distributed tracing, and much more. 
-Documentation and code samples for these new libraries can be found [here](http://aka.ms/azsdk/python/mgmt). In addition, a migration guide that shows how to transition from older versions of libraries is located [here](https://azure.github.io/azure-sdk-for-python/mgmt_preview_quickstart.html#migration-guide). 
+Documentation and code samples for these new libraries can be found [here](http://aka.ms/azsdk/python/mgmt). In addition, a migration guide that shows how to transition from older versions of libraries is located [here](https://github.com/Azure/azure-sdk-for-python/blob/master/doc/sphinx/mgmt_quickstart.rst#migration-guide).
```
Comment (johanste) [inline README.md:39]: Were these changes intentional? If so, it would probably be better with a separate PR since I don't believe they are related to the KeyVault query backup/restore status methods, right?

Comment (mccoyp) [reply README.md:39]: This was definitely unintentional -- I had to rebase to pull some changes in order to get tests to work, and those pulled commits ended up being tacked onto my most recent one. Sorting this out this morning!

**Review by chlowell — APPROVED** [review]

**Review by rllane17 — APPROVED** [review]
Good 👍

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_backup_client.py [hunk 3]
```diff
@@ -102,3 +102,35 @@ async def begin_selective_restore(
...
+        """Returns the status of a restore operation.
+
+        :param str job_id: The ID returned as part of the restore request
+        :returns: The restore operation status as a :class:`RestoreOperation`
+        :rtype: RestoreOperation
+        """
+        return await self._client.restore_status(
+            vault_base_url=self._vault_url,
+            job_id=job_id,
+            cls=RestoreOperation._wrap_generated,
```
[the remark this answers is not in the document]
Comment (rllane17) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_backup_client.py:134]: Use this SelectiveKeyRestoreOperation common first, then use
